app/code/Magento/NewsletterApi/Api/NewsletterSubscriptionRepositoryInterface.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\NewsletterApi\Api;
+
+/**
+ * Newsletter Subscription CRUD interface.
+ *
+ * @api
+ */
+interface NewsletterSubscriptionRepositoryInterface
+{
+    /**
+     * Save newsletter subscription.
+     *
+     * @param \Magento\NewsletterApi\Api\Data\NewsletterSubscriptionInterface $newsletterSubscription
+     * @return \Magento\NewsletterApi\Api\Data\NewsletterSubscriptionInterface
+     * @throws \Magento\Framework\Exception\LocalizedException
```

Comment:
Localized is too generic.
next exception could be thrown:
     * @throws \Magento\Framework\Exception\InputException
     * @throws \Magento\Framework\Exception\StateException
     * @throws \Magento\Framework\Exception\CouldNotSaveException